src/main/java/co/inventorsoft/StreamHomework.java

```diff
@@ -39,7 +56,10 @@
      * @return map {user email : user}
      */
     public Map<String, User> groupByEmail(final List<User> users) {
-        return null;
+        Map<String, User> output = new HashMap<>();
+        users.forEach((k) -> output.put(k.getEmail(), k));
```

Comment:
Good, but still, it is all about Stream API, try to find one more way how to achieve it using streams.